src/librustc/session/config.rs

```diff
@@ -501,6 +501,8 @@ options! {CodegenOptions, CodegenSetter, basic_codegen_options,
         "system linker to link outputs with"),
     link_args: Option<Vec<String>> = (None, parse_opt_list,
         "extra arguments to pass to the linker (space separated)"),
+    no_gc_sections: bool = (false, parse_bool,
```

Comment:
I'd personally prefer to call this just `gc_sections` with a default of `true`, it helps avoid double negatives later on. That way to turn it off you specify `-C gc-sections=no` or whatever we interpret as "falsey" values